apps/src/templates/census2017/CensusTeacherBanner.jsx

```diff
@@ -0,0 +1,350 @@
+                onSchoolTypeChange={this.onSchoolTypeChange.bind(this)}
+                onSchoolChange={this.onSchoolChange.bind(this)}
+                onSchoolNotFoundChange={this.onSchoolNotFoundChange.bind(this)}
+                country={this.state.country}
+                schoolType={this.state.schoolType}
+                ncesSchoolId={schoolId}
+                schoolName={this.state.schoolName}
+                schoolState={this.state.schoolState}
+                schoolZip={this.state.schoolZip}
+                schoolLocation={this.state.schoolLocation}
+                useGoogleLocationSearch={true}
+                showErrors={this.state.showSchoolInfoErrors}
+                showRequiredIndicator={true}
+              />
+            </div>
+            <div style={styles.buttonDiv}>
+              <Button onClick={this.dismissSchoolInfoForm.bind(this)} style={styles.button} color="gray" size="large" text="Dismiss" />
+              <Button onClick={this.onSchoolInfoSubmit.bind(this)} style={styles.button} size="large" text="Submit" />
+            </div>
+          </form>
+        </div>
+      );
+    } else {
+      const numHours = (this.props.question === 'how_many_20_hours') ? '20' : '10';
+      let  buttons;
+      let  footer;
+      if (this.props.selection===true) {
+        footer = (<hr/>);
+        buttons = (
+          <div style={styles.buttonDiv}>
+            <Button onClick={this.props.onDismiss} style={styles.button} color="gray" size="large" text="No thanks" />
+            <Button onClick={this.props.onSubmit} style={styles.button} size="large" text="Add my school to the map!" />
+          </div>
+        );
+      } else if (this.props.selection===false) {
+        footer = (
+          <div>
+            <hr/>
+            <p>We’d love to know more about computer science opportunities at your school. Please take our survey to increase access to Computer Science in the US.</p>
+          </div>
+        );
+        buttons = (
+          <div style={styles.buttonDiv}>
+            <Button onClick={this.props.onPostpone} style={styles.button} color="gray" size="large" text="Not now" />
+            <Button onClick={this.props.onPostpone} href={pegasus('/yourschool')} style={styles.button} size="large" text="Take the survey" />
+          </div>
+        );
+      }
+
+      let schoolName;
+
+      if (this.state.schoolDisplayName) {
+        schoolName = this.state.schoolDisplayName;
+      } else if (this.state.schoolName) {
+        schoolName = this.state.schoolName;
+      }
+
+      const schoolId = this.state.ncesSchoolId ? this.state.ncesSchoolId : this.props.ncesSchoolId;
+
+      if (schoolName) {
+        return (
+          <div style={styles.main}>
+            <form id="census-teacher-banner-form">
+              <input type="hidden" name="nces_school_s" value={schoolId}/>
+              <input type="hidden" name="school_year" value={this.props.schoolYear}/>
```

Comment:
Are these needed? See above

Reply:
I suppose if this is simpler as a form with uncontrolled elements so we don't need a bunch of state management then it makes sense. In that case, please move the whole form into a different component (or at the very least a function), and use ref to find that DOM element. 

This render method is getting a bit long and unwieldy, hard to keep track of the branches.